integration/v3_grpc_test.go

```diff
@@ -428,6 +428,11 @@ func TestV3WatchFromCurrentRevision(t *testing.T) {
 
 // TestV3WatchCancel tests Watch APIs cancellation.
 func TestV3WatchCancel(t *testing.T) {
+	testV3WatchCancel(t, 0) // synced
+	testV3WatchCancel(t, 1) // unsynced
```

Comment:
I think it'd be easier to debug test failures if this is split into `TestV3WatchCancelSynced` and `TestV3WatchCancelUnsynced`